src/app/qtlocalpeer/qtlockedfile_unix.cpp

```diff
@@ -130,6 +130,7 @@ bool QtLockedFile::unlock()
     }
 
     m_lock_mode = NoLock;
+    remove();
```

Comment:
Please don't break file handling logic. Destroying file object (as well as unlocking file) shouldn't mean removing the file. You should explicitly unlock it and then remove in QtLocalPeer destructor instead.

Reply:
Should we follow `QLockFile`? 
`~QLockFile()` will delete the lock file: https://doc.qt.io/qt-5/qlockfile.html#dtor.QLockFile

Reply:
Then it should not inherit the QFile, so as not to give the incorrect impression that it is just an extended version of the regular file handler.

Reply:
IMO, if we want to change it so deeply it can be done in separately.